src/libcore/iter/range.rs

```diff
@@ -400,6 +401,73 @@ step_integer_impls! {
     wider than usize: [u32 i32], [u64 i64], [u128 i128];
 }
 
+#[unstable(feature = "step_trait", reason = "recently redesigned", issue = "42168")]
+unsafe impl Step for char {
+    #[inline]
+    fn steps_between(&start: &char, &end: &char) -> Option<usize> {
+        let start = start as u32;
+        let end = end as u32;
+        if start <= end {
+            let count = end - start + 1;
+            if start < 0xD800 && 0xE000 <= end {
+                usize::try_from(count - 0x800).ok()
+            } else {
+                usize::try_from(count).ok()
+            }
```

Comment:
This looks wrong according to the spec for Step::steps_between.

https://github.com/rust-lang/rust/blob/45127211566c53bac386b66909a830649182ab7a/src/libcore/iter/range.rs#L31-L32

The code here does not return `Some(0)` if start == end.

Reply:
Good catch!
I was too focused on the inclusive case, I forgot `steps_between` is not the inclusive range length 😅